### PR TITLE
Feat: 5530 feature updated at inserted at properties on retrieved records

### LINKED ISSUE #5530: [FEATURE] updated_at / inserted_at properties on retrieved Records
**Why do we want this feature**

When building and maintaining datasets, it is critical to trace **when** a data point has been generated or modified. As this can be used for:

* KPI definition
* Explainability and debugging of new models
* Better understanding of drifts
* ...

**Is your feature request related to a problem? Please describe.**

When storing the records in any Argilla dataset, we can see that the following record information is available in the server:
image

Nevertheless, when loading that record using the python client, we cannot access to the `updated_at` and `inserted_at` fields. 

We checked the Record model and saw that these properties are not available in the python client.

**Describe the solution you'd like**
The properties of `updated_at` and `inserted_at` are retrieved from the server and can be retrieved from the Record model as properties.

Ideally, these two properties could also be used for filtering records.

**Describe alternatives you've considered**
Using metadata to that end. It might be a temporary fix but adds unnecessary redundancy (this information already exists in the record at server-side) and adding some server-generated timestamp will be more robust.

### PR DESCRIPTION
**Pull request by davidberenstein1957**
# Description

Closes #5530 

**Type of change**

- Improvement (change adding some improvement to an existing functionality)

**How Has This Been Tested**

**Checklist**

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

**Comment by davidberenstein1957** [comment]
@frascuchon do you think it also makes sense to update the `__repr__` of all resources to include this info?

**Comment by frascuchon** [comment]
> @frascuchon do you think it also makes sense to update the `__repr__` of all resources to include this info?

Yes, I think it makes sense

File: argilla/tests/integration/test_list_records.py [hunk 1]
```diff
@@ -106,3 +106,21 @@ def test_list_records_with_responses(client: Argilla, dataset: Dataset):
 
     assert records[1].responses["comment"][0].value == "The comment"
     assert records[1].responses["sentiment"][0].value == "negative"
+
+
+def test_list_records_with_updated_at_and_inserted_at(client: Argilla, dataset: Dataset):
```
Comment (frascuchon) [inline argilla/tests/integration/test_list_records.py:198]: Should we expose these new attributes in the `to_dict` methods? cc @burtenshaw

Comment (davidberenstein1957) [reply argilla/tests/integration/test_list_records.py:198]: for me it makes sense too.

Comment (burtenshaw) [reply argilla/tests/integration/test_list_records.py:198]: That should be ok. They will be used in `records._io._generic` and exported in all formats, so we should validate serialisation.

Comment (frascuchon) [reply argilla/tests/integration/test_list_records.py:198]: I see the `.serialize` methods already serialize these new attributes. I would leave it as is and review the logic when unifying to_dict and serialize methods: Ref #4944 

(I added a todo commit https://github.com/argilla-io/argilla/pull/5540/commits/87cb1ebc3973f4b327b2dd2dc0e76f5db563dc1f)

**Review by frascuchon — APPROVED** [review]

**Pull request merged**